packages/zoe/src/contracts/loan/borrow.js

```diff
@@ -0,0 +1,149 @@
+
+    // The value of the collateral in the Loan brand
+    const collateralPriceInLoanBrand = await E(priceOracle).getInputPrice(
+      collateralGiven,
+      loanBrand,
+    );
+    // AWAIT ///
+
+    // formula: assert collateralValue*100 >= wantedLoan*mmr
+    // TODO: assess for rounding errors
+
+    const approxForMsg = loanMath.make(
+      natSafeMath.floorDivide(natSafeMath.multiply(wantedLoan.value, mmr), 100),
+    );
+    assert(
+      loanMath.isGTE(
+        loanMath.make(
+          natSafeMath.multiply(collateralPriceInLoanBrand.value, 100),
+        ),
+        loanMath.make(natSafeMath.multiply(wantedLoan.value, mmr)),
+      ),
+      details`The required margin is approximately ${approxForMsg} but collateral only had value of ${collateralPriceInLoanBrand}`,
+    );
+
+    // Assert that the collateralGiven has not changed after the AWAIT
+    assert(
+      collMath.isEqual(
+        collateralGiven,
+        borrowerSeat.getAmountAllocated('Collateral'),
+      ),
+      `The collateral allocated changed during the borrow step, which should not have been possible`,
+    );
+
+    const { zcfSeat: collateralSeat } = zcf.makeEmptySeatKit();
+
+    // Transfer the wanted Loan amount to the collateralSeat
+    trade(
+      zcf,
+      {
+        seat: lenderSeat,
+        gains: {},
+      },
+      { seat: collateralSeat, gains: { Loan: wantedLoan } },
+    );
+
+    // Transfer *all* collateral to the collateral seat. Transfer the
+    // wanted Loan amount to the borrower.
+    trade(
+      zcf,
+      {
+        seat: collateralSeat,
+        gains: { Collateral: collateralGiven },
+      },
+      { seat: borrowerSeat, gains: { Loan: wantedLoan } },
+    );
+
+    // We now exit the borrower seat so that the borrower gets their
+    // loan. However, the borrower gets an object as their offerResult
+    // that will let them continue to interact with the contract.
+    borrowerSeat.exit();
+
+    // The liquidationTriggerValue is when the value of the collateral
+    // equals mmr percent of the wanted loan
+    // Formula: liquidationTriggerValue = (wantedLoan * mmr) / 100
+    const liquidationTriggerValue = loanMath.make(
+      natSafeMath.floorDivide(natSafeMath.multiply(wantedLoan.value, mmr), 100),
+    );
+    const debt = wantedLoan;
+
+    // TODO: calculate interest
+    // QUESTION: how is interest (aka stability fee on Maker) calculated?
+    // Maker calculates on every block?
+    const getInterest = () => loanMath.getEmpty();
+    const getDebt = () => loanMath.add(debt, getInterest());
+
+    const configWithBorrower = { ...config, getDebt, collateralSeat };
+
+    const liquidationPromise = E(priceOracle).priceWhenLT(
+      collateralGiven,
+      liquidationTriggerValue,
+    );
+
+    liquidationPromise
+      .then(({ quoteAmount }) => {
+        const expectedValueOfCollateral = quoteAmount.value[0].price;
+        return liquidate(zcf, configWithBorrower, expectedValueOfCollateral);
+      })
```

Comment:
I think when liquidiationPromise resolves, we need to recalculate, since the borrower might have added collateral. I think this turns into a method that is called to reschedule liquidation when adding or removing collateral, or when this promise resolves. 
I can't think of a way to unschedule a promise, so that means when changing collateral, we'd just be adding more triggers, and each of the triggers needs to verify that its expectations are still valid.